Products/ZenEvents/zentrap.py

```diff
@@ -760,6 +774,12 @@ class TrapDaemon(CollectorDaemon):
 
     _frameworkFactoryName = "nosip"
 
+    def __init__(self, *args, **kwargs):
+        self._trapFilter = TrapFilter()
+        zope.component.provideUtility(self._trapFilter, ICollectorEventTransformer)
+        super(TrapDaemon, self).__init__(*args, **kwargs)
+        self._trapFilter.initialize()
```

Comment:
use an initializecallback in the super constructor to init filters.  By the time the init callback is triggered you can send an event to zenoss if the filters have an error.  Be sure to send a clear event if filters are successful or no filters defined.